mautic/mautic
# added track_private_ip_ranges option to allow private IPs to be tracked

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Yes
| Automated tests included? | Yes
| Related user documentation PR URL |
| Related developer documentation PR URL |
| Issues addressed (#s or URLs) |
| BC breaks? | No
| Deprecations? | No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

We run Mautic on our local office network, and as a result, we need to be able to test tracking of IPs in our private IP range. Currently, Mautic doesn't allow this because all private IPs are converted to `127.0.0.1` in the [`IpLookupHelper::ipIsValid` method](https://github.com/mautic/mautic/blob/2.14.0/app/bundles/CoreBundle/Helper/IpLookupHelper.php#L210). Specifically, the [`FILTER_FLAG_NO_PRIV_RANGE` flag](http://php.net/manual/en/filter.filters.flags.php) passed to `filter_var` filters out IPs in the given private ranges.

This PR adds a config option named `track_private_ip_ranges` that disables the usage of the `FILTER_FLAG_NO_PRIV_RANGE` flag and therefore does not convert internal IPs to `127.0.0.1`. The default value for `track_private_ip_ranges` is `false`, making this change backwards compatible with existing installations.

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Set `track_private_ip_ranges` to `true` in `app/config/local.php`.
2. Clear the symfony cache (e.g. `rm -rf app/cache/*`).
3. Generate some tracking hits from an internal IP (e.g. `10/8`, `172.16/12`, or `192.168/16`) on the same network as the mautic installation.
4. Verify that the internal IP is correctly tracked.
5. Verify that external IP tracking still works with `track_private_ip_ranges` set to `true` or `false` (remember to clear the symfony cache after each config change).
